tests/integrations/langchain/helpers_test.py

```diff
@@ -0,0 +1,597 @@
+"""
+Unit tests for langchain integration helpers.
+
+Tests the usage metadata extraction and processing functions that handle
+different provider formats (OpenAI, Google GenAI, Google Vertex AI, Anthropic).
+"""
+
+from unittest.mock import Mock
+
+from weave.integrations.langchain.helpers import (
+    _extract_model_from_flattened_path,
+    _extract_usage_data,
+    _normalize_token_counts,
+    _reconstruct_usage_data_from_flattened,
+    _validate_usage_shape,
+)
+from weave.trace.weave_client import Call
+
+
+class TestExtractUsageData:
+    """Test the main usage data extraction function."""
+
+    def test_openai_format(self):
+        """Test OpenAI token_usage format extraction."""
+        call = Mock(spec=Call)
+        call.summary = None
+
+        output = {
+            "outputs": [
+                {
+                    "generations": [
+                        [
+                            {
+                                "message": {
+                                    "kwargs": {
+                                        "token_usage": {
+                                            "prompt_tokens": 10,
+                                            "completion_tokens": 5,
+                                            "total_tokens": 15,
+                                        }
+                                    }
+                                },
+                                "generation_info": {"model_name": "gpt-4o-mini"},
+                            }
+                        ]
+                    ]
+                }
+            ]
+        }
+
+        _extract_usage_data(call, output)
+
+        assert call.summary is not None
+        assert "usage" in call.summary
+        usage = call.summary["usage"]
+        assert "gpt-4o-mini" in usage
+        assert usage["gpt-4o-mini"]["prompt_tokens"] == 10
+        assert usage["gpt-4o-mini"]["completion_tokens"] == 5
+        assert usage["gpt-4o-mini"]["total_tokens"] == 15
```

Comment:
better to assert on the whole dict.  You will get a nicer error in pytest

Reply:
resolved. Testing the whole dict here and everywhere else